### PR TITLE
[8.0] ADD account_invoice_rounding_by_currency

### PR DESCRIPTION
**Pull request by eLBati**
Unit rounded invoice (Swedish rounding) by currency
===================================================

This module extends functionality of module Unit rounded invoice https://github.com/OCA/account-invoicing/tree/8.0/account_invoice_rounding

It allows to set, in accounting settings, a rounding precision for each currency,
such as 0.05 CHF for Swiss invoices

Configuration
=============

In Settings -> Configuration -> Accounting you will find
Currencies Rounding Rules
Set currency rule for each currency you need to handle.

- `Swedish Round globally`

  To round your invoice total amount, this option will do the adjustment in
  the most important tax line of your invoice.

- `Swedish Round by adding an invoice line`

  To round your invoice total amount, this option create a invoice line without
  taxes on it. This invoice line is tagged as `is_rounding`

### CODE REVIEW

File: account_invoice_rounding_by_currency/README.rst
Comment (elicoidal) [inline account_invoice_rounding_by_currency/README.rst]: Is it officially called "Swedish Rounding"?

File: account_invoice_rounding_by_currency/README.rst
Comment (elicoidal) [inline account_invoice_rounding_by_currency/README.rst]: please add a period

File: account_invoice_rounding_by_currency/README.rst
Comment (elicoidal) [inline account_invoice_rounding_by_currency/README.rst]: Please use enumerated list for procedures

File: account_invoice_rounding_by_currency/README.rst
Comment (elicoidal) [inline account_invoice_rounding_by_currency/README.rst]: "this option will create an invoice line without tax on it."
add a period at the end of the sentence

File: account_invoice_rounding_by_currency/demo/data.xml
Comment (elicoidal) [inline account_invoice_rounding_by_currency/demo/data.xml]: please use English in the xml files and add a translation file

File: account_invoice_rounding_by_currency/models/company.py
Comment (elicoidal) [inline account_invoice_rounding_by_currency/models/company.py]: "summed up" (2 times)

File: account_invoice_rounding_by_currency/tests/test_rounding_by_currencies.py
Comment (elicoidal) [inline account_invoice_rounding_by_currency/tests/test_rounding_by_currencies.py:1]: Please use short headers

**Review by elicoidal — CHANGES_REQUESTED** [review]
Thanks for the contribution.
Mostly small details

File: account_invoice_rounding_by_currency/README.rst
Comment (eLBati) [reply account_invoice_rounding_by_currency/README.rst]: Hi @elicoidal the term is already used here https://github.com/OCA/account-invoicing/tree/8.0/account_invoice_rounding

See https://en.wikipedia.org/wiki/Cash_rounding

File: account_invoice_rounding_by_currency/models/company.py
Comment (eLBati) [reply account_invoice_rounding_by_currency/models/company.py]: Doesn't "sum up" mean "give a summary"?

File: account_invoice_rounding_by_currency/tests/test_rounding_by_currencies.py
Comment (eLBati) [reply account_invoice_rounding_by_currency/tests/test_rounding_by_currencies.py:1]: Done

File: account_invoice_rounding_by_currency/demo/data.xml
Comment (eLBati) [reply account_invoice_rounding_by_currency/demo/data.xml]: Done

File: account_invoice_rounding_by_currency/README.rst
Comment (eLBati) [reply account_invoice_rounding_by_currency/README.rst]: Done

File: account_invoice_rounding_by_currency/README.rst
Comment (eLBati) [reply account_invoice_rounding_by_currency/README.rst]: done

File: account_invoice_rounding_by_currency/README.rst
Comment (eLBati) [reply account_invoice_rounding_by_currency/README.rst]: I don't understand this one

**Comment by eLBati** [comment]
@elicoidal thanks for the review. I made changes

File: account_invoice_rounding_by_currency/README.rst
Comment (elicoidal) [reply account_invoice_rounding_by_currency/README.rst]: I see that but it is NZ English ;)
I would suggest to switch to "Cash Rounding" which is more in line with American English used by Odoo.

File: account_invoice_rounding_by_currency/README.rst
Comment (elicoidal) [reply account_invoice_rounding_by_currency/README.rst]: To round your invoice total amount, this option create a invoice line this option will create an invoice line without tax on it.
This invoice line is tagged as `is_rounding`.
(Period=".")

File: account_invoice_rounding_by_currency/models/company.py
Comment (elicoidal) [reply account_invoice_rounding_by_currency/models/company.py]: It does but you have the other meaning as well. Check #3 here: http://www.thefreedictionary.com/sum+up
"sum up - determine the sum ofsum up - determine the sum of; "Add all the people in this town to those of the neighboring town""

Anyway, I am not here to be picky and I am not native English so it is up to you :)

**Comment by eLBati** [comment]
@elicoidal thanks, made the changes

**Comment by eLBati** [comment]
:green_heart: green now

**Comment by hurrinico** [comment]
LGTM :+1:

**Review by tafaRU — APPROVED** [review]

**Comment by tafaRU** [comment]
@elicoidal, could you please have a final review before to merge it?
Thank you in advance.

File: account_invoice_rounding_by_currency/README.rst
Comment (elicoidal) [inline account_invoice_rounding_by_currency/README.rst]: I would suggest something like: 
```
#. in Settings > Configuration > Accounting, check Currencies Rounding Rules
#. Set currency rule for each currency you need to handle:

  - `Swedish Round globally`
    To round your invoice total amount, this option will do the adjustment in
    the most important tax line of your invoice.
  - `Swedish Round by adding an invoice line`
    To round your invoice total amount, this option will create a invoice line without
    taxes on it.
    This invoice line is tagged as `is_rounding`

```

File: account_invoice_rounding_by_currency/README.rst
Comment (elicoidal) [inline account_invoice_rounding_by_currency/README.rst]: remove extra lines

File: account_invoice_rounding_by_currency/models/company.py
Comment (elicoidal) [inline account_invoice_rounding_by_currency/models/company.py]: Usually better to avoid the "!" which makes the end user uncomfortable :wink:

**Review by elicoidal — APPROVED** [review]
some unblocking details

**Comment by eLBati** [comment]
@elicoidal thanks, done

**Pull request merged**
